src/pip/_internal/pyproject.py

```diff
@@ -58,7 +58,7 @@ def load_pyproject_toml(
     has_setup = os.path.isfile(setup_py)
 
     if has_pyproject:
-        with open(pyproject_toml, encoding="utf-8") as f:
+        with open(pyproject_toml, "rb") as f:
             pp_toml = tomli.load(f)
```

Comment:
```suggestion
        with open(pyproject_toml, encoding="utf-8") as f:
            pp_toml = tomli.loads(f.read())
```

BTW, if this change is rejected, I think we should commit this change to keep existing behavior while avoiding deprecation error from Tomli.

Reply:
Let's do this.

Reply:
:+1: Pushed this change.